exoscale_auth.py

```diff
@@ -29,3 +31,55 @@ def __call__(self, request):
             'Authorization': auth,
         })
         return request
+
+class ExoscaleV2Auth(AuthBase):
+    def __init__(self, key, secret):
+        self.key = key
+        self.secret = secret.encode('utf-8')
+
+    def __call__(self, request):
+        auth_header = u'EXO2-HMAC-SHA256 credential={}'.format(self.key)
+
+        # Request method/URL path
+        msg = '{method} {path}\n'.format(
+            method=request.method,
+            path=urlparse(request.url).path)
+
+        # Request body
+        if request.body:
+            msg += body.encode('utf-8')
+        msg += '\n'
+
+        # Request query string parameters
+        # Important: this is order-sensitive, we have to have to sort
+        # parameters alphabetically to ensure signed # values match the
+        # names listed in the "signed-query-args=" signature pragma.
+        params = parse_qs(urlparse(request.url).query)
+        signed_params = sorted(params.keys())
```

Comment:
```suggestion
        signed_params = sorted(params)
```

(cosmetic - but when iterating on a dict, the `.keys()` is implied)